python/tk_desktop/rpc.py

```diff
@@ -268,3 +393,540 @@ def close(self):
+        finally:
+            self._server.socket.close()
+
+    def close(self):
+        """
+        Close the server.
+        """
+        self._is_closed = True
+
+    def is_closed(self):
+        """
+        :returns: ``True`` is the server is closed, ``False`` if not.
+        """
+        return self._is_closed
+
+    @property
+    def server_port(self):
+        """
+        :returns: Port number used by the server.
+        """
+        return self._server.server_port
+
+
+class Handler(SimpleHTTPRequestHandler):
+    """
+    Handle an HTTP request.
+    """
+
+    @property
+    def _functions(self):
+        """
+        Shorthand to access the listener's function list.
+        """
+        return self.server.listener._functions
+
+    @property
+    def _engine(self):
+        """
+        Shorthand to access the engine we're exposing methods for.
+        """
+        return self.server.listener.engine
+
+    def do_POST(self):
+        """
+        Handle request to execute a function from the listener.
+        """
+        response = None
+        try:
+            (authkey, (is_blocking, func_name, args, kwargs)) = self._read_request()
+            logger.debug("http server calling '%s(%s, %s)'" % (func_name, args, kwargs))
+
+            # Make sure the right credentials were sent.
+            # Pickling through sgtk.util.pickle converts any binary string to string
+            # so the authkey we received needs to be compared against
+            # the str version.
+            if authkey != six.ensure_str(self.server.listener.authkey):
+                raise ValueError("invalid auth key")
+
+            # Make sure the function exists.
+            if func_name not in self._functions:
+                logger.error(
+                    "unknown function call: '%s', expecting one of '%s'"
+                    % (func_name, list(self._functions))
+                )
+                raise ValueError("unknown function call: '%s'" % func_name)
+
+            # grab the function from the function table
+            func = self._functions[func_name]
+
+            # execute the function on the main thread, as it may to GUI work.
+            if is_blocking:
+                response = self._engine.execute_in_main_thread(func, *args, **kwargs)
+            else:
+                self._engine.async_execute_in_main_thread(func, *args, **kwargs)
+                response = None
+
+            logger.debug("http '%s' result: '%s" % (func_name, response))
+        except Exception as e:
+            response = e
+            # if any of the above fails send the exception back to the client
+            logger.exception("got exception during '%s" % func_name)
+        finally:
+            self._send_response(response)
+
+    def log_request(self, code=None, size=None):
+        """
+        Prevent the server from outputting to stdout every single time
+        a query is processed by the server.
```

Comment:
As I understand it, this method is overridden and intentionally left blank so that it doesn't output everytime a query is processed. 
I did a double take when looking at this, might be good to tweak the description to explicitly state that it was left empty to prevent thee server from ...